src/services/spr.ts

```diff
@@ -0,0 +1,263 @@
+import express from 'express';
+import subdomain from 'express-subdomain';
+import Dicer from 'dicer';
+import { getDuplicateCECData, getRandomCECData } from '@/database';
+import { getNEXDataByPID, getFriends } from '@/util';
+import { CECData } from '@/models/cec-data';
+import { SendMode, SPRSlot } from '@/types/common/spr-slot';
+
+const spr = express.Router();
+
+function multipartParser(request: express.Request, response: express.Response, next: express.NextFunction): void {
+	const RE_BOUNDARY = /^multipart\/.+?(?:; boundary=(?:(?:"(.+)")|(?:([^\s]+))))$/i;
+	const RE_FILE_NAME = /name="(.*)"/;
+
+	const contentType = request.header('content-type');
+
+	if (!contentType) {
+		return next();
+	}
+
+	const boundary = RE_BOUNDARY.exec(contentType);
+
+	if (!boundary) {
+		return next();
+	}
+
+	const dicer = new Dicer({ boundary: boundary[1] || boundary[2] });
+	const files: Record<string, Buffer> = {};
+
+	dicer.on('part', (part: Dicer.PartStream) => {
+		let fileBuffer = Buffer.alloc(0);
+		let fileName = '';
+
+		part.on('header', header => {
+			const contentDisposition = header['content-disposition' as keyof object];
+			const regexResult = RE_FILE_NAME.exec(contentDisposition);
+
+			if (regexResult) {
+				fileName = regexResult[1];
+			}
+		});
+
+		part.on('data', (data: Buffer | string) => {
+			if (data instanceof String) {
+				data = Buffer.from(data);
+			}
+
+			fileBuffer = Buffer.concat([fileBuffer, data as Buffer]);
+		});
+
+		part.on('end', () => {
+			files[fileName] = fileBuffer;
+		});
+	});
+
+	dicer.on('finish', function () {
+		request.files = files;
+		return next();
+	});
+
+	request.pipe(dicer);
+}
+
+spr.post('/relay/0', multipartParser, async (request, response) => {
+	if (!request.files) {
+		response.sendStatus(400);
+		return;
+	}
+
+	if (!request.pid) {
+		response.sendStatus(400);
+		return;
+	}
+
+	// * Check that the account is a 3DS and isn't banned
+	const accountData = await getNEXDataByPID(request.pid);
+	if (!accountData) {
```

Comment:
Unlike in Go, where `if` checks happen right after function calls with error returns, we prefer separating blocks/functioncalls from definitions in TS/JS

```suggestion
	const accountData = await getNEXDataByPID(request.pid);

	if (!accountData) {
```